acstools/acszpt.py

```diff
@@ -268,116 +264,97 @@ def _check_date(self, fmt='%Y-%m-%d'):
+        """Function to query API on AWS APIGateway for zeropoints for single or all
+        filters of a given ACS detector (HRC, SBC, or WFC) on a specified date.
 
         Returns
         -------
-        tab : `astropy.table.QTable` or `None`
-            If the request was successful, returns a table; otherwise, `None`.
+        table : astropy.table.table.QTable
+            If the request was successful, returns an astropy Qtable; otherwise, `None`.
 
         """
-        LOG.info('Checking inputs...')
-        valid_inputs = self._check_inputs()
-
-        if valid_inputs:
-            LOG.info(f'Submitting request to {self._url}')
-            self._submit_request()
-            if self._failed:
-                return
+        #check user input date, detector, and filter
+        bool_inputs = self._check_inputs()
+
+        if bool_inputs:
+
+        #select between all filters or a single filter depending on if user has
+        #specified a filter, and then generate a request body to send the API
+            if self.filt is None:
+                #URL to invoke all_filter API on AWS APIGateway
+                invokeURL = "https://vtbopx9sf3.execute-api.us-east-1." \
+                                "amazonaws.com/main/all-filter-resource"
+                #Generate a request body to send the API
+                body = '{"date": "%s", "detector": "%s"}' %(self.date, self.detector)
+            else:
+                #URL to invoke single_filter API on AWS APIGateway
+                invokeURL = "https://vtbopx9sf3.execute-api.us-east-1." \
+                                "amazonaws.com/main/single-filter-resource"
+                #Generate a request body to send the API
+                body = '{"date": "%s", "detector": "%s", "filter": "%s"}' %(self.date, self.detector, self.filt)
+
+            #send request to APIGateway with try/except clauses
+            help_desk_string = "\nIf this error persists, please contact the HST Help Desk at \n" \
+            "https://stsci.service-now.com/hst"
+
+            request_exception = True
+
+            try:
+                response = requests.post(invokeURL, data = body)
+                response.raise_for_status()
+
+            except requests.exceptions.HTTPError:
+                err_string = "HTTP Error to AWS API Gateway:" + help_desk_string
+                print(err_string)
+            except requests.exceptions.ConnectionError:
+                err_string = "Error Connecting to AWS API Gateway:" + help_desk_string
+                print(err_string)
+            except requests.exceptions.Timeout:
+                print("Timeout Error to AWS API Gateway:" + help_desk_string)
+            except requests.exceptions.RequestException:
+                print("Request Exception Error to AWS API Gateway:" + help_desk_string)
+            else: 
+                request_exception = False
+
+            if request_exception:
+                sys.exit()
```

Comment:
Why `sys.exit()` here instead of just letting the exception be raised properly above?

Reply:
I think @jryon and I decided to keep the error message clean for the end-user

Reply:
I'd rather you reraise an Exception with clearer message than to exit the Python session completely.